rbac/common/proposal/proposal_state.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2018 Contributors to Hyperledger Sawtooth
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+import logging
+from rbac.addressing import addresser
+from rbac.common import protobuf
+from rbac.common.manager.base_state import BaseState
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ProposalState(BaseState):
+    def __init__(self):
+        BaseState.__init__(self)
+
+    @property
+    def name(self):
+        return "proposal"
+
+    @property
+    def container_proto(self):
+        return protobuf.proposal_state_pb2.ProposalsContainer
+
+    @property
+    def state_proto(self):
+        return protobuf.proposal_state_pb2.Proposal
+
+    def address(self, object_id, target_id=None):
+        """Make an address for the given user_id"""
+        return addresser.make_proposal_address(
+            object_id=object_id, related_id=target_id
+        )
+
+    def make(self, object_id):
+        """Makes the state storage object"""
+        raise NotImplementedError("Class must implement this method")
+
+    def create(self, signer_keypair, message, object_id=None, target_id=None):
```

Comment:
This appears to be duplicated from base state and can be removed

Reply:
what line are you commenting on?

Reply:
The create method